bpf/bpf_overlay.c

```diff
@@ -323,6 +329,49 @@ int tail_handle_ipv4(struct __ctx_buff *ctx)
 					      CTX_ACT_DROP, METRIC_INGRESS);
 	return ret;
 }
+
+#ifdef ENABLE_VTEP
+/*
+ * ARP responder for ARP requests from VTEP
+ * Respond to remote VTEP endpoint with cilium_vxlan MAC
+ */
+__section_tail(CILIUM_MAP_CALLS, CILIUM_CALL_ARP)
+int tail_handle_arp(struct __ctx_buff *ctx)
+{
+	union macaddr mac = NODE_MAC;
+	union macaddr smac;
+	__be32 sip;
+	__be32 tip;
+	__u32 monitor = 0;
+	int i;
+	struct remote_endpoint_info *info;
+	struct bpf_tunnel_key key = {};
+
+	if (unlikely(ctx_get_tunnel_key(ctx, &key, sizeof(key), 0) < 0))
+		return DROP_NO_TUNNEL_KEY;
+	if (key.tunnel_id != WORLD_ID)
+		return DROP_INVALID_VNI;
```

Comment:
my original intention is to block  ARP broadcasting/flooding from non-VTEP device, but yes, it could interfere with node or cluster-internal traffic if these traffics require ARP. I can remove this check